### PR TITLE
Bar signs now rename the area, doors and some wall items

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Bartenders can now rename the generic area "Bar" to the bar name of their choice with the use of their bar sign. This will rename air alarms, doors and APCs, so everyone knows what you're calling your bar this shift. (The bar sign chosen is also now a statistic that's tracked!)
/:cl:

Metastation now only has one bar sign.
Bar signs start blank.
Bar signs no longer require bartender ID to repair.
Emagging now only sets the bar sign (and bar name) to the Syndi Cat, rather than changing the Barsign to syndicate access only and making them impossible to repair.
Only machinery that is normally renamed when the area is renamed will have their name changed, this means only doors originally called "Bar" will be renamed.

Area renames are split into separate procs, so we can use their renaming of machinery features.

### CODE REVIEW

**Comment by tuypo1** [comment]
I think this is pretty cool

**Comment by XDTM** [comment]
Welcome to Xenobiology Lab, customers get a free slime teleported onto them

**Comment by ExcessiveUseOfCobblestone** [comment]
Maid Cafe

**Comment by PKPenguin321** [comment]
Area checks really need to use `initial ` so this doesn't break stuff

**Comment by subject217** [comment]
MAID CAFE TIME BABYYY

File: code/game/objects/structures/barsigns.dm [hunk 1]
```diff
@@ -1,45 +1,54 @@
...
 
 /obj/structure/sign/barsign/obj_break(damage_flag)
 	if(!broken && !(flags_1 & NODECONSTRUCT_1))
 		broken = TRUE
 
 /obj/structure/sign/barsign/deconstruct(disassembled = TRUE)
-	new /obj/item/stack/sheet/metal (loc, 2)
-	new /obj/item/stack/cable_coil (loc, 2)
+	new /obj/item/stack/sheet/metal(loc, 2)
+	new /obj/item/stack/cable_coil(loc, 2)
```
Comment (Cyberboss) [inline code/game/objects/structures/barsigns.dm]:
```suggestion
	new /obj/item/stack/cable_coil(drop_location(), 2)
```

File: code/game/objects/structures/barsigns.dm [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
+			return set_sign(new_sign)
 
 /obj/structure/sign/barsign/obj_break(damage_flag)
 	if(!broken && !(flags_1 & NODECONSTRUCT_1))
 		broken = TRUE
 
 /obj/structure/sign/barsign/deconstruct(disassembled = TRUE)
-	new /obj/item/stack/sheet/metal (loc, 2)
-	new /obj/item/stack/cable_coil (loc, 2)
+	new /obj/item/stack/sheet/metal(loc, 2)
```
Comment (Cyberboss) [inline code/game/objects/structures/barsigns.dm]:
```suggestion
	new /obj/item/stack/sheet/metal(drop_location(), 2)
```

File: tgstation.dme [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 // BEGIN_INCLUDE
 #include "_maps\_basemap.dm"
+#include "_maps\metastation.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: .

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**